### PR TITLE
chore: use the project's typescript version in vscode

### PR DESCRIPTION
**Pull request by JammingBen**
At least for me this causes much less issues in vscode. Otherwise it would use typescript `5.3` which is not yet compatible with Web.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Technical debt
- [ ] Tests

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -13,7 +13,6 @@
   },
   "editor.formatOnSave": true,
   "eslint.format.enable": true,
-  "eslint.packageManager": "pnpm",
```
Comment (JammingBen) [inline .vscode/settings.json:16]: Fyi this is deprecated and is now being detected automatically.

File: .vscode/settings.json [hunk 2]
```diff
@@ -27,5 +26,5 @@
   "cucumberautocomplete.steps": [
     "tests/e2e/cucumber/steps/**/*.ts"
   ],
-  "cucumberautocomplete.strictGherkinCompletion": true
```
Comment (dschmidt) [inline .vscode/settings.json:30]: Why was this removed?

Comment (JammingBen) [reply .vscode/settings.json:30]: Because I messed up 🤡

**Review by dschmidt — APPROVED** [review]
